compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -114,6 +114,20 @@ static std::string translateModuleToISA(llvm::Module &module,
   return targetISA;
 }
 
+// Modified from lib/Target/AMDGPU/AMDGPUAttributor.cpp.
+// Adds argument hints to preload kernel arguments to SGPRs.
+// TODO: Query max number of user SGPRs from target machine.
+static void addPreloadKernArgHint(llvm::Function *F) {
+  constexpr unsigned long maxSGPRs = 16;
+  for (unsigned i = 0; i < std::min(F->arg_size(), maxSGPRs); ++i) {
```

Comment:
This broke a Windows (post-submit) build: https://github.com/openxla/iree/actions/runs/7733161138/job/21084668684#step:8:8343

```
FAILED: compiler/plugins/target/ROCM/CMakeFiles/iree_compiler_plugins_target_ROCM_ROCM.objects.dir/ROCMTarget.cpp.obj 
C:\ProgramData\Chocolatey\bin\ccache C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1437~1.328\bin\HostX64\x64\cl.exe  /nologo /TP  -IC:\a\iree\iree -IC:\a\iree\iree\build-windows -IC:\a\iree\iree\third_party\llvm-project\llvm\include -IC:\a\iree\iree\build-windows\llvm-project\include -IC:\a\iree\iree\third_party\llvm-project\mlir\include -IC:\a\iree\iree\build-windows\llvm-project\tools\mlir\include -IC:\a\iree\iree\third_party\llvm-project\lld\include -IC:\a\iree\iree\build-windows\llvm-project\tools\lld\include -IC:\a\iree\iree\compiler\src -IC:\a\iree\iree\build-windows\compiler\src -IC:\a\iree\iree\llvm-external-projects\iree-dialects\include -IC:\a\iree\iree\build-windows\llvm-external-projects\mlir-iree-dialects\include -IC:\a\iree\iree\runtime\src -IC:\a\iree\iree\build-windows\runtime\src -IC:\a\iree\iree\build-windows\runtime\src\iree\base\internal\flatcc -IC:\a\iree\iree\build-windows\runtime\src\iree\schemas -external:IC:\a\iree\iree\third_party\flatcc\include -external:W0 /DWIN32 /D_WINDOWS /EHsc /Z7 /O2 /Ob1  -std:c++17 -MD /DWIN32_LEAN_AND_MEAN /DNOMINMAX /D_USE_MATH_DEFINES /D_CRT_SECURE_NO_WARNINGS /D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES /D_SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING /GR- /bigobj /Zc:preprocessor /W3 /wd4200 /wd4018 /wd4146 /wd4244 /wd4267 /wd4005 /wd4065 /wd4141 /wd4624 /wd4576 /wd5105 -IC:/a/iree/iree/third_party/flatcc/include/ -IC:/a/iree/iree/third_party/flatcc/include/flatcc/reflection/ /showIncludes /Focompiler\plugins\target\ROCM\CMakeFiles\iree_compiler_plugins_target_ROCM_ROCM.objects.dir\ROCMTarget.cpp.obj /Fdcompiler\plugins\target\ROCM\CMakeFiles\iree_compiler_plugins_target_ROCM_ROCM.objects.dir\ /FS -c C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): error C2672: 'std::min': no matching overloaded function found
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\utility(78): note: could be '_Ty std::min(std::initializer_list<_Elem>)'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: '_Ty std::min(std::initializer_list<_Elem>)': expects 1 arguments - 2 provided
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\utility(75): note: or       '_Ty std::min(std::initializer_list<_Elem>,_Pr)'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: '_Ty std::min(std::initializer_list<_Elem>,_Pr)': could not deduce template argument for 'std::initializer_list<_Elem>' from 'size_t'
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\utility(68): note: or       'const _Ty &std::min(const _Ty &,const _Ty &) noexcept(<expr>)'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: 'const _Ty &std::min(const _Ty &,const _Ty &) noexcept(<expr>)': template parameter '_Ty' is ambiguous
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: could be 'unsigned long'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: or       'size_t'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: 'const _Ty &std::min(const _Ty &,const _Ty &) noexcept(<expr>)': could not deduce template argument for 'const _Ty &' from 'const unsigned long'
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\utility(58): note: or       'const _Ty &std::min(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)'
C:\a\iree\iree\compiler\plugins\target\ROCM\ROCMTarget.cpp(122): note: 'const _Ty &std::min(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)': expects 3 arguments - 2 provided
```

(check that your data types all match / cast them to a single type before using `std::min`)

Reply:
yeah, types have to match (Function::arg_size is `unsigned`, not `unsigned long`)

Reply:
Wish we had a generic `take_front(Range, N)` in llvm. There's `drop_begin` already, but I don't like that it would require more arithmetic.

Reply:
@benvanik @ScottTodd Fix: https://github.com/openxla/iree/pull/16283